core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -1179,4 +1238,61 @@ public void setNext(int next) {
             cachedNext = next;
         }
     }
+
+    public static class ShareFetchPartitionData {
```

Comment:
maybe `private`, incase required in tests then `default`

Reply:
converting to private at the moment since we don't have any unit tests.